### PR TITLE
Adds initial yesterday count support

### PR DESCRIPTION
**Pull request by rcrichton**
@kaweesi I haven't tested this at all as I don't have a test system so it may be broken, but could you have a look through this and see what you think. This is how I'm thinking of doing counts for the day prior to when the report is executed.

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/systemmonitor/export/DHISGenerateDataValueSetSchemas.java [hunk 1]
```diff
@@ -174,6 +176,10 @@ public static JSONObject generateRwandaSPHEMTDHISDataValueSets() {
 					+ DHISMapping.getDHISMappedObjectValue("DATA-ELEMENT_totalEncounters") + "\", \"period\": \""
 					+ dFormat.format(today) + "\", \"orgUnit\": \"" + dhisOrganizationUnitUid + "\", \"value\": "
 					+ encounterTotal + "}";
+			String newEncountersDataElement = "{ \"dataElement\": \""
```
Comment (kaweesi) [inline api/src/main/java/org/openmrs/module/systemmonitor/export/DHISGenerateDataValueSetSchemas.java]: we probably need to pass/add this to a json parser and add its json to jsonDataValueSets before its tapped into jsonToBePushed

Comment (rcrichton) [inline api/src/main/java/org/openmrs/module/systemmonitor/export/DHISGenerateDataValueSetSchemas.java]: I've updated the code to add this change.

File: api/src/main/java/org/openmrs/module/systemmonitor/api/db/hibernate/HibernateSystemMonitorDAO.java [hunk 2]
```diff
@@ -119,6 +119,26 @@ public String getToday() {
...
+
+		return sdf.format(calendar.getTime());
+	}
+
+	@Override
+	public String getYesterdayEndDate() {
+		Calendar calendar = Calendar.getInstance(Context.getLocale());
+
+		resetDateTimes(calendar);
+		calendar.add(Calendar.MILLISECOND, -1);
```
Comment (rcrichton) [inline api/src/main/java/org/openmrs/module/systemmonitor/api/db/hibernate/HibernateSystemMonitorDAO.java]: @kaweesi what do you think about this end date calculation? I subtracted a millisecond to make sure we don't get results at midnight today, but I see this isn't does on the other end dates. Is it not required do you think?

Comment (kaweesi) [inline api/src/main/java/org/openmrs/module/systemmonitor/api/db/hibernate/HibernateSystemMonitorDAO.java]: I had actually not reviewed this, am thinking that we probably need to first of all subtract one from DAY_OF_MONTH to fall onto yesterday. am seemingly agreeing that after going backward by day, we can then just subtract one from milli_secs, i will actually first need to check this out, you could probably test this whole method in a unit test

**Comment by kaweesi** [comment]
This works pretty well, thanks @rcrichton 👍

**Pull request merged**